llvm/test/CodeGen/AMDGPU/sdwa-peephole-cndmask-fail.ll

```diff
@@ -0,0 +1,40 @@
+; RUN: llc %s -march=amdgcn -mcpu=gfx1030 -o - 2>&1 | FileCheck %s
+
+; In this test, V_CNDMASK_B32_e64 gets converted to V_CNDMASK_B32_e32,
+; but the expected conversion to SDWA does not occur.  This led to a
+; compilation error, because the use of $vcc in the resulting
+; instruction must be fixed to $vcc_lo for wave32 which only happened
+; after the full conversion to SDWA.
+
+
+; CHECK-NOT: {{.*}}V_CNDMASK_B32_e32{{.*}}$vcc
+; CHECK-NOT: {{.*}}Bad machine code: Virtual register defs don't dominate all uses
```

Comment:
-NOT checks are next to useless, particularly for the error message. Just rely on the test not failing with the verifier

Reply:
I don't think they are useless. They serve as documentation - perhaps a comment would be just as good for this purpose - and they can improve the error message that you will see in case of a regression.

Reply:
Can you also add this case to the mir test

Reply:
I think I cannot add it to the existing test file for "wave32" which uses "--run-pass=si-peephole-sdwa". The mir parser cannot read the mir that llc outputs with "--stop-before=si-peephole-sdwa" because it contains instructions with implicit "vcc_lo" operands. The parser insists on those being "vcc" instead because it does not take the wave32/wave64 distinction into account while checking the correctness of the implicit operands.

Reply:
Added it now after manually editing the mir output.

Reply:
These are useless and should be removed. It's broken as soon as any detail of the output changes, and it is not the point of the test. This is not a verifier test, at most comment it's avoiding a verifier failure